engine/classes/Elgg/PluginHooksService.php

```diff
@@ -25,11 +25,35 @@ public function trigger($hook, $type, $params = null, $returnvalue = null) {
 		$hooks = $this->getOrderedHandlers($hook, $type);
 		
 		foreach ($hooks as $callback) {
+
+			$callback_copy = $callback;
+
+			if (is_string($callback)
+					&& false === strpos($callback, '::')
+					&& !function_exists($callback)
+					&& class_exists($callback)) {
+
+				$class = ltrim($callback, '\\');
+
+				$cached = _elgg_services()->config->get("object:$class");
+				if (!$cached) {
+					$factory = _elgg_services()->config->get("class_factory:$class");
+					if ($factory) {
+						$cached = call_user_func($factory, $class);
+					} else {
+						$cached = new $class();
```

Comment:
I think this should be the only thing we support right now. Then when we offer DI we can use `$injector->instantiate($class);` or whatever the equivalent is.